src/style/style.js

```diff
@@ -622,6 +622,10 @@ class Style extends Evented {
         this._order.splice(index, 1);
 
         const newIndex = before ? this._order.indexOf(before) : this._order.length;
+        if (before && newIndex === -1) {
+            this.fire('error', { message: new Error(`Layer with id "${before}" does not exist on this map.`)});
```

Comment:
The key here should be `error` since you're passing an error object, not a message.

Reply:
Oh, I see it was mixed up in the original code.

Reply:
I think you're right, the key here should be `error` as well as here https://github.com/mapbox/mapbox-gl-js/pull/5679/files#diff-0fcf9ef37a1614185cad0dcbd5ec5529L571

Reply:
Shall I change it to a `{error: ....}`?

Reply:
yes please @jingsam and it would be great if you could change the it to `{error: new Error...` on line 571 as well. thank you!